src/backend/gporca/libgpopt/src/xforms/CXformUtils.cpp

```diff
@@ -1340,19 +1338,14 @@ CXformUtils::PexprLogicalDMLOverProject(CMemoryPool *mp,
 	IMDId *rel_mdid = ptabdesc->MDId();
 	CExpression *pexprProject = NULL;
 	CColRef *pcrAction = NULL;
-	CColRef *pcrOid = NULL;
+	CColRef *pcrOid = pcrTableOid;
 
 	if (ptabdesc->IsPartitioned())
 	{
 		// generate a PartitionSelector node which generates OIDs, then add a project
 		// on top of that to add the action column
 		CExpression *pexprSelector = PexprLogicalPartitionSelector(
 			mp, ptabdesc, colref_array, pexprChild);
-		if (CUtils::FGeneratePartOid(ptabdesc->MDId()))
```

Comment:
Deletion of generation pcrOid for Insert operation (and using the pcrOid from the function argument (in case of insert it's always NULL)) leads to SIGSEGV at next test cases of `CDMLTest`.

- src/backend/gporca/data/dxl/minidump/Insert-Parquet-Partitioned.mdp
- src/backend/gporca/data/dxl/minidump/Insert-AO-Partitioned.mdp

For now pcrOid is `NULL` and the problem occurrs at this [code](https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/src/backend/gporca/libgpopt/src/operators/CPhysicalDML.cpp#L531-L537).

Previously for the first test case Oid was generated due to FGeneratePartOid check passed (here is a [function](https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/src/backend/gporca/libgpopt/src/base/CUtils.cpp#L4410-L4429))
- first test case is called on parquet table.
-  second test case [uses special hint](https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/data/dxl/minidump/Insert-AO-Partitioned.mdp#L14) while adb-6.x [COptTask sets this hint to int_max](https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/src/backend/gpopt/utils/COptTasks.cpp#L397).

Seems this if should be returned